src/newgrf_config.cpp

```diff
@@ -537,51 +515,24 @@ bool GRFFileScanner::AddFile(const std::string &filename, size_t basepath_length
 	/* Abort if the user stopped the game during a scan. */
 	if (_exit_game) return false;
 
-	GRFConfig *c = new GRFConfig(filename.c_str() + basepath_length);
-
-	bool added = true;
+	bool added = false;
+	auto c = std::make_unique<GRFConfig>(filename.c_str() + basepath_length);
+	GRFConfig *grfconfig = c.get();
 	if (FillGRFDetails(*c, false)) {
-		if (_all_grfs == nullptr) {
-			_all_grfs = c;
-		} else {
-			/* Insert file into list at a position determined by its
-			 * name, so the list is sorted as we go along */
-			GRFConfig **pd, *d;
-			bool stop = false;
-			for (pd = &_all_grfs; (d = *pd) != nullptr; pd = &d->next) {
-				if (c->ident.grfid == d->ident.grfid && c->ident.md5sum == d->ident.md5sum) added = false;
-				/* Because there can be multiple grfs with the same name, make sure we checked all grfs with the same name,
-				 *  before inserting the entry. So insert a new grf at the end of all grfs with the same name, instead of
-				 *  just after the first with the same name. Avoids doubles in the list. */
-				if (StrCompareIgnoreCase(c->GetName(), d->GetName()) <= 0) {
-					stop = true;
-				} else if (stop) {
-					break;
-				}
```

Comment:
Just for the next time reviewing: sorting is *also* done by `DoScanNewGRFFiles`, making all attempts to sort here kinda pointless.